src/main/java/seedu/smarthomebot/logic/commands/ListCommand.java

```diff
@@ -69,31 +80,39 @@ private CommandResult listAppliance() throws LocationNotFoundException, EmptyApp
             NoApplianceInLocationException {
         String outputResult;
         ArrayList<Appliance> outputApplianceList = applianceList.getAllAppliance();
-
         if (filteredLocation.equals("")) {
-            if (outputApplianceList.size() == 0) {
-                throw new EmptyApplianceListException();
-            }
-            String header = MESSAGE_LIST_APPLIANCES;
-            outputResult = displayOutput(header, outputApplianceList);
+            outputResult = listAllAppliances(outputApplianceList);
         } else {
-            ArrayList<Appliance> filterApplianceList =
-                    (ArrayList<Appliance>) outputApplianceList.stream()
-                            .filter((s) -> s.getLocation().equals(filteredLocation))
-                            .collect(toList());
-
-            if (filterApplianceList.isEmpty()) {
-                if (locationList.isLocationCreated(filteredLocation)) {
-                    throw new NoApplianceInLocationException();
-                }
-                throw new LocationNotFoundException();
-            }
-            String header = ("Here are the Appliances in \"" + filteredLocation + "\"");
-            outputResult = displayOutput(header, filterApplianceList);
+            outputResult = listApplianceByLocation(outputApplianceList);
         }
         return new CommandResult(outputResult);
     }
 
+    private String listAllAppliances(ArrayList<Appliance> outputApplianceList) throws EmptyApplianceListException {
+        if (outputApplianceList.size() == 0) {
+            throw new EmptyApplianceListException();
+        }
+
+        return displayOutput(MESSAGE_LIST_APPLIANCES, outputApplianceList);
+    }
+
+    private String listApplianceByLocation(ArrayList<Appliance> outputApplianceList)
+            throws LocationNotFoundException, NoApplianceInLocationException {
+        ArrayList<Appliance> filterApplianceList =
+                (ArrayList<Appliance>) outputApplianceList.stream()
+                        .filter((s) -> s.getLocation().equals(filteredLocation))
+                        .collect(toList());
+
+        if (filterApplianceList.isEmpty()) {
+            if (locationList.isLocationCreated(filteredLocation)) {
+                throw new NoApplianceInLocationException();
+            }
+            throw new LocationNotFoundException();
+        }
+        String header = ("Here are the Appliances in \"" + filteredLocation + "\"");
+        return displayOutput(header, filterApplianceList);
```

Comment:
I think the displayOutput should store string, then return with string is better. It will be much clearer in the implementation.

Reply:
So you are saying:
String displayOutput = header + filterApplianceList ;
return displayOutput;
?

Reply:
Yes

Reply:
This cannot be done as displayOutput takes in a header and a ApplianceList